src/meta_schedule/feature_extractor/per_block_feature.cc

```diff
@@ -705,44 +658,55 @@ void Feature::SubFeature::SetFeature(const LoopNest& loop_nest, int64_t cache_li
 }
 
 Feature::Feature(const BlockRealizeNode* realize, const LoopNest& loop_nest,
-                 int64_t cache_line_bytes, IntVec* for_touched_bytes,
-                 ForBufferMap<IntVec>* buffer_touched_under_loop, arith::Analyzer* analyzer,
-                 std::vector<const BlockRealizeNode*>* scopes) {
-  int n_loops = loop_nest.loops.size();
+                 int64_t cache_line_bytes, const BlockRealizeNode* parent_scope,
+                 IntVec* for_touched_bytes, arith::Analyzer* analyzer) {
   // Step 0. Initialize data structures
-  this->Init(realize, n_loops, analyzer, scopes);
+  this->Init(realize, parent_scope, analyzer);
   // Step 1. Calculate region-related feature
-  this->SetRegion(loop_nest, for_touched_bytes, buffer_touched_under_loop, analyzer);
+  std::vector<IntVec> buffer_touched_under_loop = this->SetRegion(loop_nest);
   // Step 2. Calculate stride-related feature
-  for (auto& feature : sub_features) {
+  for (SubFeature& feature : sub_features) {
     feature.SetStride(loop_nest, analyzer);
   }
   // Step 3. Calculate reuse-related feature
-  int64_t top_loop_touch_bytes = 0.0;
-  if (n_loops > 0) {
-    for (const SubFeature& feature : sub_features) {
-      int64_t bytes = feature.buffer->dtype.bytes();
-      int64_t n_buffer = feature.loop_accessed_numel[0].size();
-      top_loop_touch_bytes += bytes * n_buffer;
-    }
-  }
-  for (auto& feature : sub_features) {
-    feature.SetReuse(loop_nest, top_loop_touch_bytes, *buffer_touched_under_loop);
+  for (SubFeature& feature : sub_features) {
+    feature.SetReuse(loop_nest, buffer_touched_under_loop, sub_features);
```

Comment:
Could we just sub_features to static data member or tweak around the function declaration so that we do not pass in sub_features as an argument?

Reply:
It's a good point though, but given it's passed by constant reference (i.e. underlyingly it's a pointer) it should be fine